src/app/components/content-finder-page/content-finder-page.component.ts

```diff
@@ -142,26 +144,40 @@ export class ContentFinderPageComponent implements OnInit {
          
          //populate a filter object with the params we just extracted
          let filter: Filter = new Filter(
-            title, format, moduleIdNumbers
+            title, formats, moduleIdNumbers
          );
 
          this.sendSearch(filter);
       }
    }
 
+   toggleFormat(format : string){
+
+
+      
+      if(this.selFormat.includes(format)){
+         if(this.selFormat.length==1){
```

Comment:
Is this if statement necessary? The toastr info states that the user must include at least one format type. However, this if statement shows that exactly one format type is selected. Isn't this good? Why is the user being told to select a format, if one format is already selected?

Reply:
If this is necessary, shouldn't the logic check that the selFormat.length == 0 instead of 1?

Reply:
Because they are deselecting.  If there is only one format type, and they are trying to deselect it there will be none.

Reply:
if(this.selFormat.includes==format) means that the button being clicked is active and therefor they are trying to make it inactive
length==1 means it is the only format left currently

Reply:
I see, I suppose that makes sense. But from a user experience standpoint, not being able to deselect something until you select something else isn't ideal. Preferably, you would temporarily allow no formats to be selected, and only send an error if they try to submit with no formats.

Reply:
Do you want me to change it then?

Reply:
If you have time, then yes.